cdap-master/src/main/java/co/cask/cdap/master/startup/MasterStartupTool.java

```diff
@@ -74,6 +78,25 @@ public static void main(String[] args) {
       throw Throwables.propagate(e);
     }
 
+    // check for potential port conflicts
```

Comment:
logic should go in ConfigurationCheck, though we actually can't check these things because its possible for all these services to be running on different machines.  The only thing we can really check is that these ports are valid numbers.